### PR TITLE
Blobbernauts are healed quicker by cores and nodes

### PR DESCRIPTION
**Pull request by ghost**
i hate sitting for 3-5 minutes waiting to heal as a naut

:cl: More Robust Than you
balance: Blobbernauts are healed quicker by cores and nodes
/:cl:

### CODE REVIEW

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 1]
```diff
@@ -244,6 +244,10 @@
 			if(overmind)
 				I.color = overmind.blob_reagent_datum.complementary_color
 			flick_overlay_view(I, src, 8)
+		if(factory && (locate(/obj/structure/blob/core) in range(2, src)))
+			adjustHealth(-maxHealth*0.1)
+		if(factory && (locate(/obj/structure/blob/node) in range(2, src)))
```
Comment (RemieRichards) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: >calling ```range(2, src)``` three times in the same life tick

HECK NO.

**Review by RemieRichards — DISMISSED** [review]

File: code/game/gamemodes/blob/blobs/blob_mobs.dm (hunk 1 above)
Comment (MoreRobustThanYou) [reply code/game/gamemodes/blob/blobs/blob_mobs.dm]: Ok, i've moved it to a single call per life tick

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 2]
```diff
@@ -221,6 +221,7 @@
 	see_in_dark = 8
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
 	var/independent = FALSE
+	var/list/blobs_in_area
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: w h y

Comment (MoreRobustThanYou) [reply code/game/gamemodes/blob/blobs/blob_mobs.dm]: it was either this or calling range 3 times per life tick

Comment (ChangelingRain) [reply code/game/gamemodes/blob/blobs/blob_mobs.dm]: don't put this list as a var on the mob jesus christ

Comment (MoreRobustThanYou) [reply code/game/gamemodes/blob/blobs/blob_mobs.dm]: ok

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 3]
```diff
@@ -244,6 +245,10 @@
 			if(overmind)
 				I.color = overmind.blob_reagent_datum.complementary_color
 			flick_overlay_view(I, src, 8)
+		if(factory && (locate(/obj/structure/blob/core) in blobs_in_area))
+			adjustHealth(-maxHealth*0.1)
+		if(factory && (locate(/obj/structure/blob/node) in blobs_in_area))
+			adjustHealth(-maxHealth*0.05)
```
Comment (Cyberboss) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: These 4 lines should be tabbed in once more because damagesources is guaranteed to be > 0 if it runs

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MoreRobustThanYou** [comment]
tbh i can't remember why i closed this

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 4]
```diff
@@ -229,13 +229,18 @@
...
 			return // strong independent blobbernaut that don't need no blob
 		var/damagesources = 0
-		if(!(locate(/obj/structure/blob) in range(2, src)))
+		if(!(locate(/obj/structure/blob) in blobs_in_area))
 			damagesources++
 		if(!factory)
 			damagesources++
+		if(factory && (locate(/obj/structure/blob/core) in blobs_in_area))
+			adjustHealth(-maxHealth*0.1)
+		if(factory && (locate(/obj/structure/blob/node) in blobs_in_area))
```
Comment (Cyberboss) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]:
```
if(!factory)
    damagesources++
else
    if(locate(/obj/structure/blob/core) in blobs_in_area)
        adjustHealth(-maxHealth*0.1)
    if(locate(/obj/structure/blob/node) in blobs_in_area)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 5]
```diff
@@ -229,13 +229,29 @@
...
 		if(independent)
 			return // strong independent blobbernaut that don't need no blob
 		var/damagesources = 0
-		if(!(locate(/obj/structure/blob) in range(2, src)))
+		if(!(locate(/obj/structure/blob) in blobs_in_area))
 			damagesources++
 		if(!factory)
 			damagesources++
+		else
+			if(factory && (locate(/obj/structure/blob/core) in blobs_in_area))
```
Comment (Cyberboss) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: the point of the else is to remove the factory checks

File: code/game/gamemodes/blob/blobs/blob_mobs.dm [hunk 6]
```diff
@@ -229,13 +229,29 @@
...
+				adjustHealth(-maxHealth*0.1)
+				var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(src)) //hello yes you are being healed
+				if(overmind)
+					H.color = overmind.blob_reagent_datum.complementary_color
+				else
+					H.color = "#000000"
+			if(factory && (locate(/obj/structure/blob/node) in blobs_in_area))
```
Comment (Cyberboss) [inline code/game/gamemodes/blob/blobs/blob_mobs.dm]: you fucking meme

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/blob/blobs/blob_mobs.dm (hunk 5 above)
Comment (MoreRobustThanYou) [reply code/game/gamemodes/blob/blobs/blob_mobs.dm]: ohh

**Comment by MoreRobustThanYou** [comment]
@tgstation/commit-access please review

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**